internal/controller/proxmoxcluster_controller.go

```diff
@@ -177,6 +198,11 @@ func (r *ProxmoxClusterReconciler) reconcileNormal(ctx context.Context, clusterS
 		return res, nil
 	}
 
+	if err := r.reconcileNormalCredentialsSecret(ctx, clusterScope); err != nil {
+		conditions.MarkFalse(clusterScope.ProxmoxCluster, infrav1alpha1.ProxmoxClusterReady, infrav1alpha1.ProxmoxUnreachableReason, clusterv1.ConditionSeverityError, err.Error())
```

Comment:
here you will need to check for `apierrors.IsNotFound()`
and throw an error in the cluster cause the credentials secret is not found.

Reply:
Done. Thanks!